test/unittests/test_xml_upload_stash.py

```diff
@@ -0,0 +1,86 @@
+# pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
+
+from unittest import TestCase
+
+import pytest
+
+from dsp_tools.models.value import KnoraStandoffXml
+from dsp_tools.utils import xml_upload_stash
+
+
+class TestXMLUploadStash(TestCase):
+    def test_find_ids_referenced_in_salsah_links_one_link(self) -> None:
+        one_link_KnoraStandoffXml = KnoraStandoffXml(
+            xmlstr=(
+                '<resource label="r1_label" restype="r1_restype" id="r1_id" permissions="res-default">'
+                '<text-prop name=":hasRichtext"><text permissions="res-default" encoding="xml">'
+                '<a class="salsah-link" href="IRI:r2_id:IRI">r2_id</a>'
+                "</text></text-prop></resource>"
+            )
+        )
+        returned_set = one_link_KnoraStandoffXml.find_ids_referenced_in_salsah_links()
+        self.assertEqual({"r2_id"}, returned_set)
+
+    def test_find_ids_referenced_in_salsah_links_three_links(self) -> None:
```

Comment:
Wow, these small methods are much more readable than the parametrized ones that I'm used to. This will be a lesson for myself, because now is the first time I realize the benefits!